### PR TITLE
Brings back rspec_retry; Creates helper method for flaky spec #9811

### PR DESCRIPTION
**Pull request by filipefurtad0**
#### What? Why?

**The problem**

Some specs are flaky because the behavior of the app is flaky itself ([example below](https://github.com/openfoodfoundation/openfoodnetwork/pull/9831#issuecomment-1285312478)).

We've figured the best way is address this and avoid a flood of flaky specs is to bring back `rspec_retry` and mark the flaky example with `retry:`.

**Which settings should we choose?**

Here I've chosen to mark the two examples in this PR with `retry: 2, retry_wait: 2`. 

If we're conservative and consider a fail rate of 0.1, then the probability of failing a second rerun should be 0.1^2 (1%). With a default wait time between re-runs of 2s this would add 4 seconds wait time (I think the wait time is added each time the spec is ran, also at the first attempt) and the spec run time. 

Something to keep in mind is that if these settings are changed on the globally (through a global variable or on `base_spec_helper.rb`) then the retries will run on **all** failing specs - this is a bug, mentioned on the gem documentation - and something we which we want to avoid. I've also tested this locally and could confirm it.

The PR also creates a click_select2 helper method to improve flakiness (issue #9811 and #9810).

Brings back `rspec_retry`, marks flaky specs with retry option.
Replaces the helper method with more find/click routine.

#### What should we test?

- we should get a consistent, green build
- only the specs marked with `retry: 2, retry_wait: 2` should re-run, when failing, waiting 2 seconds preceding each retry.

#### Release notes

Changelog Category: Technical changes

The title of the pull request will be included in the release notes.

### CODE REVIEW

**Comment by filipefurtad0** [comment]
There are some inconsistencies around the Clear All button / search function :sob: 

- upon page load

image

- after selecting some filters and clearing them

image

Notice the date has not changed :see_no_evil:

**Comment by filipefurtad0** [comment]
Having this into consideration I'd keep this PR and not spend more time on this particular example. But I think this flaky behavior explains the massive increase in test failures on this page (and the retry). Maybe we could consider re-adding the `retry` option only for this spec?

File: spec/system/admin/bulk_order_management_spec.rb
Comment (mkllnk) [inline spec/system/admin/bulk_order_management_spec.rb:480]: This is now repeated a lot. We should introduce a better helper for this. Maybe like this?
```suggestion
          click_on_select2("order_cycle_filter", oc1.name)
```

**Review by mkllnk — CHANGES_REQUESTED** [review]
I would merge this if the specs were 100% but activating these imperfect specs would just make our build even more flaky.

**Comment by filipefurtad0** [comment]
The failing above is spec is #8701.

I've created a helper now and passed `96 of 100 passed (96%)`. I think this could be one such case of adding the `rspec-retry`.

**Comment by mkllnk** [comment]
> 96 of 100 passed (96%)

Just a warning about this value: this is the average but we don't know the variance. The same code may result in 85%, then 96% and then 90% when running the script several times. Of course, the more often we run it, the more confident we can be but seeing an improvement from 85 to 90 does not mean that we actually improved anything. It could just be fluke.

**Comment by mkllnk** [comment]
> I think this could be one such case of adding the rspec-retry.

Agree, can you go ahead?

**Comment by filipefurtad0** [comment]
> The same code may result in 85%, then 96% and then 90% when running the script several times.

Yes absolutely, the main message is that it's still failing but that the pass rate is quite high. These numbers should be taken with a grain of salt, but we can still use them do decide how many retries we should have per flaky example.

I've re-introduced `rspec-retry` and updated the PR name and description.

File: spec/system/admin/bulk_order_management_spec.rb
Comment (jibees) [inline spec/system/admin/bulk_order_management_spec.rb:489]: We probably should remove the two previous lines right? 

```
find("#s2id_distributor_filter").click
find(".select2-result-label", text: d2.name.to_s).click
```

**Review by jibees — APPROVED** [review]
All clear thanks! 
Should we open an epic that contains an issue for each spec that used this retry?

File: spec/system/admin/bulk_order_management_spec.rb
Comment (filipefurtad0) [reply spec/system/admin/bulk_order_management_spec.rb:489]: Argh missed those. Fixing.

Comment (filipefurtad0) [reply spec/system/admin/bulk_order_management_spec.rb:487]: Done, thank you!

File: spec/system/admin/bulk_order_management_spec.rb
Comment (filipefurtad0) [reply spec/system/admin/bulk_order_management_spec.rb]: Done on 3447959a061eb1d3867eee1e5f55556dcdfcfb63 :+1:

**Comment by filipefurtad0** [comment]
> Should we open an epic that contains an issue for each spec that used this retry?

Epic created [here](https://github.com/openfoodfoundation/openfoodnetwork/issues/9871).

Will then create an issue for `bulk_order_management_spec.rb` once we merge this PR.

File: spec/base_spec_helper.rb
Comment (mkllnk) [inline spec/base_spec_helper.rb:65]: We don't need to set this. The default is `1` which means that the example is tried once. The name is confusing.

> Note that currently this is a 'try' count. If increased from the default of 1, all examples will be retried. We plan to fix this as a breaking change in version 1.0.

https://github.com/NoRedInk/rspec-retry#configuration

File: spec/system/admin/bulk_order_management_spec.rb
Comment (mkllnk) [inline spec/system/admin/bulk_order_management_spec.rb:477]: Why the wait? Did you see any changes in success here?

**Review by mkllnk — CHANGES_REQUESTED** [review]
Excellent. Just minor comments. The less config we use, the less we have to maintain when the gem updates.

File: spec/system/admin/bulk_order_management_spec.rb
Comment (filipefurtad0) [reply spec/system/admin/bulk_order_management_spec.rb]: I can't say I did - I did not investigate whether it played a role, for this spec. I've removed this option now.

Thanks!

**Comment by filipefurtad0** [comment]
Moved code review :+1

**Review by mkllnk — APPROVED** [review]
Great!

**Pull request merged**
